File: Solutions/Python/Rearrange_a_string.py
class Solution:
    def arrangeString(self, s):
        # code here
        num = []
        arr = []
        for i in s:
            if i.isdigit():
                num.append(int(i))
            else:
                arr.append(i)
        x = sum(num)
        x = [x]
        arr.sort()
        if x != [0]:
            arr.extend(x)
        return "".join(map(str,arr))

File: Solutions/Java/Rearrange_a_string.java
class Solution
{
    public String arrangeString(String s) {
        //code here.
        String arr[] = new String[26];
        Arrays.fill(arr,"");
        int sum = 0;
        String str = "";
        for (int i=0; i<s.length(); i++) {
            if (s.charAt(i)>='A' && s.charAt(i)<='Z') {
                arr[s.charAt(i)-'A'] += s.charAt(i);
            }
            else {
                sum += s.charAt(i)-'0';
            }
        }
        for (int i=0; i<26; i++) {
            str += arr[i];
        }
        
        if (sum!=0) {
            str+=sum;
        }
        return str;
    }
}
